packages/next/client/script.tsx

```diff
@@ -119,7 +121,11 @@ export function handleClientScriptLoad(props: ScriptProps) {
   const { strategy = 'afterInteractive' } = props
   if (strategy === 'lazyOnload') {
     window.addEventListener('load', () => {
-      requestIdleCallback(() => loadScript(props))
+      requestIdleCallback(() => {
+        nextjsStartTransition(() => {
+          loadScript(props)
+        })
+      })
     })
```

Comment:
Per React documents:

> `React.startTransition` lets you mark updates inside the provided callback as transitions.

However, `loadScript` is not an "update" function as it will not cause a state update or triggers a re-render (`loadScript` manipulate DOM directly, bypassing React completely). 

Besides, currently `next/script` component is not concurrent rendering safe at all (see #40025), and calling concurrent rendering API will cause the component (and its children) to opt-in concurrent rendering. If you add `startTransition` inside `next/script`, you may trigger many hidden race conditions.

Reply:
> Per React documents:
> 
> > `React.startTransition` lets you mark updates inside the provided callback as transitions.
> 
> However, calling `loadScript` will not trigger an "update" (`loadScript` will not cause a state update or triggers a re-render, as it manipulates DOM directly, bypassing React completely).
> 
> Besides, currently `next/script` component is not concurrent rendering safe at all (see #40025), and calling concurrent API will cause the component (and its children) to opt-in concurrent rendering. If you add `startTransition` inside `next/script`, you may trigger many hidden race conditions.

Ohh ok & tks for explanation 💪🏽🤝

Reply:
> Per React documents:
> 
> > `React.startTransition` lets you mark updates inside the provided callback as transitions.
> 
> However, calling `loadScript` will not trigger an "update" (`loadScript` will not cause a state update or trigger a re-render, as it manipulates DOM directly, bypassing React completely).
> 
> Besides, currently `next/script` component is not concurrent rendering safe at all (see #40025), and calling concurrent API will cause the component (and its children) to opt-in concurrent rendering. If you add `startTransition` inside `next/script`, you may trigger many hidden race conditions.

https://github.com/vercel/next.js/pull/40191 can be implemented now ?

Reply:
Still no.

> Per React documents:
> 
> > `React.startTransition` lets you mark updates inside the provided callback as transitions.
> 
> However, calling `loadScript` will not trigger an "update" (`loadScript` will not cause a state update or trigger a re-render, as it manipulates DOM directly, bypassing React completely).

Reply:
Ok :)